control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go

```diff
@@ -89,6 +101,39 @@ func ReconcileStatefulSet(ss *appsv1.StatefulSet, p *EtcdParams) error {
 	return nil
 }
 
+func buildEtcdInitContainer(p *EtcdParams) func(c *corev1.Container) {
+	return func(c *corev1.Container) {
+		// FIXME: etcdctl restore is deprecated but the etcd container doesn't have etcdutl
+		script := `
+mkdir -p /var/lib/data
+[ "$(ls -A /var/lib/data)" ] && echo "/var/lib/data not empty, not restoring snapshot" && exit 0
+HOSTNAME_SUFFIX=${HOSTNAME/-/_}
+RESTORE_URL_VAR="RESTORE_URL_${HOSTNAME_SUFFIX^^}"
+RESTORE_URL=${!RESTORE_URL_VAR}
```

Comment:
it'd be possibly easier to maintain if we put this script in a .sh file so syntax errors can be detected immediately, that then we sprintf here?

Reply:
Sure will do - I just copied the existing pattern from `buildEtcdContainer` but I guess that's only running a single command

To clarify, how do you envisage syntax errors being detected?  (I don't see shellcheck enabled atm)

Reply:
Now moved into a separate script